bin/oneoff/aichat_requests_to_csv

```diff
@@ -0,0 +1,168 @@
+      if script_level
+        # This handles chat requests that occurred on standard levels in a progression
+        path = script_level.path
+      else
+        # This handles chat requests that occurred on a sublevel
+        parent_levels = BubbleChoice.parent_levels(level.name)
+        parent_levels_in_script = parent_levels.filter do |pl|
+          pl.script_levels.any? {|sl| sl.script_id == row.script_id}
+        end
+
+        if parent_levels_in_script
+          parent_level = parent_levels_in_script.first
+          sublevel_position = parent_level.sublevel_position(level)
+          path = parent_level.build_script_level_path(
+            parent_level.script_levels.first,
+            {sublevel_position: sublevel_position}
+          )
+        end
+      end
+
+      if execution_status == SharedConstants::AI_REQUEST_EXECUTION_STATUS[:USER_PROFANITY]
+        # Parse response for openai/comprehend/blocklist/webpurify details for user message.
+        # There is no model response recorded if user message was flagged.
+        # After https://github.com/code-dot-org/code-dot-org/pull/61095, response was updated to JSON.
+        # Only handle json responses.
+        if valid_json?(response)
+          user_safety_layer_response = JSON.parse(response).deep_symbolize_keys
+          user_blocked_by = user_safety_layer_response[:blocked_by]
+          details = user_safety_layer_response[:details]
+          user_blocked_by_details = get_blocked_by_details(user_blocked_by, details)
+        end
+      elsif execution_status == SharedConstants::AI_REQUEST_EXECUTION_STATUS[:MODEL_PROFANITY]
+        # Model response is recorded in addition to blocked by service details.
+        # After https://github.com/code-dot-org/code-dot-org/pull/61095, response was updated to JSON.
+        # Only handle json responses.
+        if valid_json?(response)
+          model_safety_layer_response = JSON.parse(response).deep_symbolize_keys
+          model_blocked_by = model_safety_layer_response[:blocked_by]
+          details = model_safety_layer_response[:details]
+          model_blocked_by_details = get_blocked_by_details(model_blocked_by, details)
+        end
+      elsif execution_status == SharedConstants::AI_REQUEST_EXECUTION_STATUS[:SUCCESS]
+        # Model response is recorded.
+        model_message_text = response
+      end
+      # No model response recorded when unexpected error or user input too large.
+
+      base_row = [
+        row.id,
+        row.user_id,
+        user.username,
+        user.name,
+        path,
+        row.created_at,
+        row.updated_at,
+      ]
+
+      status =
+        case execution_status
+        when SharedConstants::AI_REQUEST_EXECUTION_STATUS[:SUCCESS] || SharedConstants::AI_REQUEST_EXECUTION_STATUS[:MODEL_PROFANITY]
+          "okay"
+        when SharedConstants::AI_REQUEST_EXECUTION_STATUS[:FAILURE]
+          "unexpected error"
```

Comment:
In the failure case, the `response` will contain the exception message. Do we want to include that here?

Reply:
Oh yeah - definitely is helpful to extract msg. Will update!